### PR TITLE
ci: refactor ci and compose

### LINKED ISSUE #138: refactor: compose and actions
After #137 got merged, to remove duplicate codes and reduce maintenance:
1. move compose yaml to one single folder, and use environmental variable to control it
2. use matrix in github actions to achieve test for different environment

Ref: https://github.com/Kong/docker-kong

### PR DESCRIPTION
**Pull request by Yiyiyimu**
Signed-off-by: yiyiyimu <wosoyoung@gmail.com>

fix #138

Also I'm not sure the reason we need to get image tarball, so remove it. If there is need, I'll revert it back.

### CODE REVIEW

File: .github/workflows/apisix-docker-test.yaml [hunk 1]
```diff
@@ -10,17 +10,32 @@ on:
...
+        if: ${{ matrix.platform == 'alpine-local' }}
         run: |
           git clone http://github.com/apache/apisix apisix-local
-          export APISIX_PATH="./apisix-local"
-          make build-on-alpine-local
-          docker-compose -f ./alpine-local/docker-compose.yaml up -d
+
+      - name: Build and run
+        run: |
+          make build-on-${{ matrix.platform }}
```
Comment (gxthrj) [inline .github/workflows/apisix-docker-test.yaml:34]: 👍

File: Makefile [hunk 2]
```diff
@@ -16,23 +16,23 @@
 #
 default: help
 
-APISIX_VERSION ?= 2.3
+APISIX_VERSION ?= latest
```
Comment (gxthrj) [inline Makefile:19]: The version should be 2.3
We discussed before and decided to modify this version number to trigger CI after each APISIX release.

Comment (Yiyiyimu) [reply Makefile]: Fixed.

Comment (Yiyiyimu) [reply Makefile]: BTW, I removed the image tarball creation and upload, do we still need them

Comment (gxthrj) [reply Makefile]: We still need them

Comment (Yiyiyimu) [reply Makefile]: @gxthrj fixed~ PTAL

**Review by gxthrj — APPROVED** [review]
LGTM

File: .github/workflows/apisix-docker-test.yaml [hunk 3]
```diff
@@ -10,16 +10,33 @@ on:
...
 jobs:
   build:
+    strategy:
+      fail-fast: false
+      matrix:
+        platform:
+          - alpine
+          - centos
+          - alpine-local
+    env:
```
Comment (Yiyiyimu) [inline .github/workflows/apisix-docker-test.yaml:20]: wait.... format error causing CI not triggered

**Review by Yiyiyimu — COMMENTED** [review]
~NOT MERGE for now~

Fixed. Ready to merge

File: .github/workflows/apisix-docker-test.yaml (hunk 3 above)
Comment (gxthrj) [reply .github/workflows/apisix-docker-test.yaml]: :)

Comment (Yiyiyimu) [reply .github/workflows/apisix-docker-test.yaml]: My bad🤣

**Review by gxthrj — APPROVED** [review]

**Review by starsz — APPROVED** [review]

**Review by tokers — APPROVED** [review]

**Pull request merged**